src/System.Windows.Extensions/src/System/Drawing/ImageConverter.cs

```diff
@@ -63,5 +75,75 @@ public override PropertyDescriptorCollection GetProperties(ITypeDescriptorContex
         }
 
         public override bool GetPropertiesSupported(ITypeDescriptorContext context) => true;
+
+        private unsafe Stream GetBitmapStream(byte[] rawData)
+        {
+            Debug.Assert(rawData != null, "rawData is null.");
+            try
+            {
+                fixed (byte* pByte = rawData)
+                {
+                    IntPtr addr = (IntPtr)pByte;
+
+                    if (addr == IntPtr.Zero)
+                    {
+                        return null;
+                    }
+
+                    // This will be pHeader.signature, but we try to
+                    // do this first so we avoid building the structure when we shouldn't
+                    if (rawData.Length <= sizeof(OBJECTHEADER) || Marshal.ReadInt16(addr) != 0x1c15)
+                    {
+                        return null;
+                    }
+
+                    // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push imagesinto the DB.
+                    OBJECTHEADER pHeader = (OBJECTHEADER)Marshal.PtrToStructure(addr, typeof(OBJECTHEADER));
+
+                    // "PBrush" should be the 6 chars after position 12 as well.
+                    if (rawData.Length <= pHeader.headersize + 18 || Encoding.ASCII.GetString(rawData, pHeader.headersize + 12, 6) != "PBrush")
+                    {
+                        return null;
+                    }
+
+                    // We can safely trust that we've got a bitmap.
+                    byte[] searchArray = Encoding.ASCII.GetBytes("BM");
+
+                    // search for "BM" in the data which is the start of our bitmap data.
+                    // 18 is from (12+6) above.
+                    for (int i = pHeader.headersize + 18; i < pHeader.headersize + 510 && i + 1 < rawData.Length; i++)
+                    {
+                        if (searchArray[0] == pByte[i] &&
+                            searchArray[1] == pByte[i + 1])
+                        {
+                            return new MemoryStream(rawData, i, rawData.Length - i);
+                        }
+                    }
+                }
+            }
+            catch (OutOfMemoryException) // This exception may be caused by creating a new MemoryStream.
+            {
+            }
+            catch (ArgumentException) // This exception may be caused by Marshal.PtrToStructure.
+            {
+            }
+
+            return null;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct OBJECTHEADER
```

Comment:
This should probably go in its own file, and possibly in a file in `src\Common\src\Interop\Windows\ole32` (?). Not sure whether we are putting code there that's only used by one library, but this seems like a fairly general struct that should go there.

Reply:
Oops, I was mixing it up with OBJECT_HEADER. This I guess is more obscure. Maybe it's best left here.

https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-oleds/3c695db0-943f-48a0-b134-939af3b3a4ca

Reply:
For the record, here is the actual definition of this:

``` C++
struct OBJECTHEADER {
    WORD        typ;
    WORD        cbHdr;
    LONG        lobjTyp;
    WORD        cbName;
    WORD        cbClass;
    WORD        ibName;
    WORD        ibClass;
    POINT       ptSize;
    WCHAR        rgch[];
};

// lobjTyp:
#define OT_LINK             1L
#define OT_EMBEDDED         2L
#define OT_STATIC           3L
```

Old stuff here. :P